Source/Applications/NullApplicationStructure.cs

```diff
@@ -12,5 +12,40 @@ public class NullApplicationStructure : IApplicationStructure
     {
         /// <inheritdoc/>
         public IApplicationStructureFragment Root => NullApplicationStructureFragment.Instance;
+
```

Comment:
Same points about CompareTo implementation and Equals.

When I have a null implementation, I tend to instantiate a single instance and make that static readonly.  You could always make the constructor private if you wanted to make sure there were no other instances instantiated.